### ISSUE TITLE
Cases - State Machine

### ISSUE THREAD

**Issue by StephenOTT** [DigitalState/Platform-legacy#15]
Re-Introduce Cases State Machine:

- [ ] Case State: Active, InActive
- [ ] Admin UI: Add DataGrid on Case View showing the History of Case State Transitions
    - review: https://github.com/orocrm/platform/blob/master/src/Oro/Bundle/WorkflowBundle/Resources/doc/reference/workflow/workflow-entities.md#workflow-transition-record for more info about ability to show State transitions in a datagrid

**Comment by sadortun** [comment]
Have a look at the Oro Workflow, it's quite powerfull !!

**Comment by StephenOTT** [comment]
@sadortun agreed.  We were originally waiting to use the workflows(state machine) until the Web API for Workflows was introduced as part of OroPlatform 2.0
